### ISSUE TITLE
Kontena Master doesn't report loss of contact with Kontena Agent/Nodes

### ISSUE THREAD

**Issue by four43** [kontena/kontena#167]
When I was dealing with some odd VPN issues, #150, I eventually found that my issue was due to the Kontena Agent/Node not being able to contact the Kontena Master and vice versa. I had accidentally removed TCP port 8080 from our Security Group and the Kontena servers were no longer able to talk to each other. 

```
$ kontena node list
Name                           OS                                       Driver          Labels                         Status    
ip-10-0-0-99                   Ubuntu 14.04.3 LTS (3.13.0-63-generic)   aufs            -                              online
```

A status of `online` is unexpected as the Node/Master haven't been in communication over port 8080. Grid list wasn't of much help either:

```
$ kontena grid list
Name                           Nodes    Services     Users     
mygrid *                       1        0            1   
```

The issue became more apparent when modifying things, for example: 

```
~$ kontena vpn delete
{"code":503,"message":"Node unavailable","backtrace":null}
```

I would think this would be reproducible by blocking TCP port 8080 between node containers and running the previous commands.

Thanks guys,
-Seth

**Comment by jakolehm** [comment]
Yes, there seems to be an edge case when connection state is not detected. Currectly connection state is checked from websocket connection between server and node. But if connection state gets out of sync and node is not coming back, master does nothing... 

I think that only way to fix this is some kind of cleanup routine in master that checks `last_seen_at` timestamp. @nevalla @miskun what do you think?

**Comment by jakolehm** [comment]
@four43 workaround for this is to remove "ghost" node from master manually:

```
$ kontena node remove ip-10-0-0-99
```

**Comment by miskun** [comment]
@jakolehm I believe this issue (among others) will be addressed and fixed in #110?

**Comment by jakolehm** [comment]
I'm pretty sure that this issue has already been fixed in 0.9.
